frontend/cypress/integration/common/istio_config.ts

```diff
@@ -620,27 +630,34 @@ function waitUntilConfigIsVisible(attempt: number, crdInstanceName: string, crdN
   cy.get('tr')
     .each($row => {
       const dataTestAttr = $row[0].attributes.getNamedItem('data-test');
+      const hasNA = $row[0].innerText.includes('N/A');
       if (dataTestAttr !== null) {
-        if (dataTestAttr.value === `VirtualItem_Ns${namespace}_${crdName}_${crdInstanceName}`) {
-          found = true;
+        if (dataTestAttr.value === `VirtualItem_Ns${namespace}_${crdName}_${crdInstanceName}` && !hasNA) {
+          // Check if the health status icon is correct
+          cy.get(`[data-test=VirtualItem_Ns${namespace}_${crdName}_${crdInstanceName}] span.pf-v5-c-icon`)
+            .should('be.visible')
+            .then(icon => {
+              const colorVar = `--pf-v5-global--${healthStatus}-color--100`;
+              const color = getComputedStyle(icon[0]).getPropertyValue(colorVar);
+              if (color) {
+                found = true;
+              }
+            });
         }
       }
     })
     .then(() => {
       if (!found) {
-        cy.wait(40000);
-        waitUntilConfigIsVisible(attempt - 1, crdInstanceName, crdName, namespace);
+        cy.wait(10000);
+        waitUntilConfigIsVisible(attempt - 1, crdInstanceName, crdName, namespace, healthStatus);
       }
     });
 }
 
 Then(
   'the {string} {string} of the {string} namespace should have a {string}',
   (crdInstanceName: string, crdName: string, namespace: string, healthStatus: string) => {
-    waitUntilConfigIsVisible(3, crdInstanceName, crdName, namespace);
-    cy.get(`[data-test=VirtualItem_Ns${namespace}_${crdName}_${crdInstanceName}] span.pf-v5-c-icon`)
-      .should('be.visible')
-      .hasCssVar('color', `--pf-v5-global--${healthStatus}-color--100`);
+    waitUntilConfigIsVisible(3, crdInstanceName, crdName, namespace, healthStatus);
```

Comment:
What happens if the health status is incorrect? Does the function run indefinitely? I'm not sure if Cypress has a time limit per scenario. If not, I think we should add a `retries` parameter to the function (waiting until the config is visible for up to X seconds) and perform a final check.

Reply:
The function `waitUntilConfigIsVisible` has the retry parameter, which is set 3 here, this function was modified to not only check the config visibility but also the healthStatus icon of config. So it should fail after 3rd attempt (x10000ms).

Reply:
You are right, I did not see the `attempt` input parameter. However, when `attempt === 0` the function returns and the test scenario is ok because there is no final validation

Reply:
Good catch, have added error when retries reach 0.